### ISSUE TITLE
Use ephemeral ports in the Kitura tests

### ISSUE THREAD

**Issue by pushkarnk** [IBM-Swift/Kitura#1361]
## Context and Description
The tests that create an HTTP server always bind to and listen on port number 8080. Though this is fine when we run the tests sequentially, we run into port collisions when we try to run them in parallel (using swift test --parallel). We must ideally be using ephemeral ports to avoid this issue.

The other option is to configure SO_REUSEPORT by default, but with this tests that verify the presence and absence of SO_REUSEPORT become irrelevant.

Ephemeral ports may lie in the range starting 16000 and up to 65535. Using the sysctl command on Linux and macOS, it is possible to limit them to a given range. In the context of Kitura, limiting them to a range so that the port number can be accommodated in an UInt16 essential because ClientRequest.Options.port is an `UInt16`.

**Comment by weissi** [comment]
you could just bind to port `0` and the OS will give you a free port. Then you just need to pass the actually bound port to your client to connect to. That's what we always do in NIO, see [for example here](https://github.com/apple/swift-nio/blob/master/Tests/NIOHTTP1Tests/HTTPServerClientTest.swift#L401-L413). We bind to `0` and then connect to `serverChannel.localAddress` which will contain the port number the OS picked automatically. That way you'll never get any issues with running tests in parallel or other programs having bound a certain port (and you don't need to implement any retrying). For TestFoundation I made a similar [change](https://github.com/apple/swift-corelibs-foundation/pull/1185/files) a while ago.

**Comment by weissi** [comment]
@pushkarnk port numbers in TCP are 16 bits wide so a `UInt16` will be able to hold all valid port numbers, no need to restrict anything.

![screenshot 2018-11-16 at 12 19 11 pm](https://user-images.githubusercontent.com/624238/48620950-d48bcd00-e999-11e8-8151-d747ea492846.png)

**Comment by pushkarnk** [comment]
@weissi I'm sorry for the wrong information, `port` is an `Int16`  (not `UInt16`) [in ClientRequest](http://ibm-swift.github.io/Kitura-net/Classes/ClientRequest/Options.html#/s:9KituraNet13ClientRequestC7OptionsO4portAEs5Int16VcAEmF). This is causing overflow issues if the port number exceeds 2^15 − 1.

**Comment by pushkarnk** [comment]
PR: https://github.com/IBM-Swift/Kitura/pull/1382

**Comment by weissi** [comment]
@pushkarnk are you sure you don't want to (until you can fix the type) use `0...32767` as the actual values and map -32768...-1 to 32768...65535. 

```swift
func portNumberFromKituraPort(_ port: Int16) -> UInt16 {
    return UInt16(bitPattern: Int16(port))
}
```

should do it.

**Comment by pushkarnk** [comment]
Merged #1382
